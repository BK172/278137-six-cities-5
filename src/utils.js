import _ from "lodash";

const MarkupCitiesList = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`,
];

export const ResponseType = {
  SUCCESS: `SUCCESS`,
  ERROR: `ERROR`,
};

export const HttpCode = {
  UNAUTHORIZED: 401,
};

export const AppRoute = {
  MAIN: `/`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  ROOM: `/offer/:id`,
};

export const APIRoute = {
  OFFERS: `/hotels`,
  LOGIN: `/login`,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const getElementWidthByRating = (rating) => Math.round(rating) * 20;

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const initialSortingType = `popular`;
export const SortingTypes = {
  'popular': `Popular`,
  'to-high': `Price: low to high`,
  'to-low': `Price: high to low`,
  'top-rated': `Top rated first`
};

export const orderCitiesByList = (cities) => {
  const orderedCities = [];

  MarkupCitiesList.forEach((cityName) => {
    const city = cities.find((city) => city.name === cityName);
    city && orderedCities.push(city);
  });

  return orderedCities;
};

export const getCitiesFromOffersList = (offers) => {
  const cities = offers.map((offer) => offer.city);
  return _.uniqWith(cities, _.isEqual);
};

export const citiesAdapter = (city) => {
  return {
    name: city.name,
    coordinates: [city.location.latitude, city.location.longitude],
    zoom: city.location.zoom,
  };
};

export const offersAdapter = (offer) => {
  return {
    bedrooms: offer.bedrooms,
    city: citiesAdapter(offer.city),
    description: offer.description,
    facilities: offer.goods,
    owner: {
      id: offer.host.id,
      avatar: offer.host.avatar_url,
      name: offer.host.name,
      super: offer.host.is_pro,
    },
    offerId: offer.id,
    photos: offer.images,
    favorite: offer.is_favorite,
    premium: offer.is_premium,
    coordinates: [offer.location.latitude, offer.location.longitude],
    zoom: offer.location.zoom,
    guests: offer.max_adults,
    image: offer.preview_image,
    price: offer.price,
    rating: offer.rating,
    title: offer.title,
    type: offer.type,
  };
};
