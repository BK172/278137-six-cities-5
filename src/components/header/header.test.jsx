import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {MemoryRouter} from "react-router-dom";
import configureStore from "redux-mock-store";
import Header from "./header";
import {makeInitialStateMock, mockAuthInfo} from "../../utils";

describe(`Header component test`, () => {
  const store = configureStore()(makeInitialStateMock());

  it(`Should render Header correctly if NO_AUTH`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter>
              <Header
                authStatus={`NO_AUTH`}
                authInfo={mockAuthInfo}
                onSignOutAction={()=>{}}
              />
            </MemoryRouter>
          </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`Should render Header correctly if AUTH`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter>
              <Header
                authStatus={`AUTH`}
                authInfo={mockAuthInfo}
                onSignOutAction={()=>{}}
              />
            </MemoryRouter>
          </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
