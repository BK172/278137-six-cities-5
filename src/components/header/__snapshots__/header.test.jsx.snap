// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component test Should render Header correctly if AUTH 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height={41}
            src="img/logo.svg"
            width={81}
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <a
              className="header__nav-link header__nav-link--profile"
              href="/favorites"
              onClick={[Function]}
            >
              <div
                className="header__avatar-wrapper user__avatar-wrapper"
                style={
                  Object {
                    "backgroundImage": "url(https://assets.htmlacademy.ru/intensives/javascript-3/avatar/1.jpg)",
                  }
                }
              />
              <span
                className="header__user-name user__name"
              >
                qwerty@qwerty.ru
              </span>
            </a>
          </li>
        </ul>
        <div
          className="header__signout"
        >
          <button
            className="header__signout-button"
            onClick={[Function]}
            title="Sign out"
            type="button"
          >
            <span
              className="header__signout-button-icon"
            />
          </button>
        </div>
      </nav>
    </div>
  </div>
</header>
`;

exports[`Header component test Should render Header correctly if NO_AUTH 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height={41}
            src="img/logo.svg"
            width={81}
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <a
              className="header__nav-link header__nav-link--profile"
              href="/favorites"
              onClick={[Function]}
            >
              <div
                className="header__avatar-wrapper user__avatar-wrapper"
                style={
                  Object {
                    "backgroundImage": "url(https://assets.htmlacademy.ru/intensives/javascript-3/avatar/1.jpg)",
                  }
                }
              />
              <span
                className="header__user-name user__name"
              >
                qwerty@qwerty.ru
              </span>
            </a>
          </li>
        </ul>
        <div
          className="header__signout"
        >
          <button
            className="header__signout-button"
            onClick={[Function]}
            title="Sign out"
            type="button"
          >
            <span
              className="header__signout-button-icon"
            />
          </button>
        </div>
      </nav>
    </div>
  </div>
</header>
`;
